tests/interlocked_win32_ut/interlocked_win32_ut.c

```diff
@@ -0,0 +1,708 @@
+MOCKABLE_FUNCTION(, BOOLEAN, mock_InterlockedCompareExchange128, volatile LONG64*, destination, LONG64, exchange_high, LONG64, exchange_low, LONG64*, comperand_result);
+MOCKABLE_FUNCTION(, SHORT, mock_InterlockedCompareExchange16, volatile SHORT*, destination, SHORT, exchange, SHORT, comperand);
+MOCKABLE_FUNCTION(, LONG64, mock_InterlockedCompareExchange64, volatile LONG64*, destination, LONG64, exchange, LONG64, comperand);
+MOCKABLE_FUNCTION(, void*, mock_InterlockedCompareExchangePointer, void * volatile*, destination, void*, exchange, void*, comperand);
+MOCKABLE_FUNCTION(, LONG, mock_InterlockedDecrement, volatile LONG*, addend);
+MOCKABLE_FUNCTION(, SHORT, mock_InterlockedDecrement16, volatile SHORT*, addend);
+MOCKABLE_FUNCTION(, LONG64, mock_InterlockedDecrement64, volatile LONG64*, addend);
+MOCKABLE_FUNCTION(, LONG, mock_InterlockedExchange, volatile LONG*, target, LONG, value);
+MOCKABLE_FUNCTION(, SHORT, mock_InterlockedExchange16, volatile SHORT*, target, SHORT, value);
+MOCKABLE_FUNCTION(, LONG64, mock_InterlockedExchange64, volatile LONG64*, target, LONG64, value);
+MOCKABLE_FUNCTION(, char, mock_InterlockedExchange8, volatile char*, target, char, value);
+MOCKABLE_FUNCTION(, LONG, mock_InterlockedExchangeAdd, volatile LONG*, addend, LONG, value);
+MOCKABLE_FUNCTION(, LONG64, mock_InterlockedExchangeAdd64, volatile LONG64*, addend, LONG64, value);
+MOCKABLE_FUNCTION(, void*, mock_InterlockedExchangePointer, void* volatile*, target, void*, value);
+MOCKABLE_FUNCTION(, LONG, mock_InterlockedIncrement, volatile LONG*, addend);
+MOCKABLE_FUNCTION(, SHORT, mock_InterlockedIncrement16, volatile SHORT*, addend);
+MOCKABLE_FUNCTION(, LONG64, mock_InterlockedIncrement64, volatile LONG64*, addend);
+MOCKABLE_FUNCTION(, LONG, mock_InterlockedOr, volatile LONG*, destination, LONG, value);
+MOCKABLE_FUNCTION(, SHORT, mock_InterlockedOr16, volatile SHORT*, destination, SHORT, value);
+MOCKABLE_FUNCTION(, LONG64, mock_InterlockedOr64, volatile LONG64*, destination, LONG64, value);
+MOCKABLE_FUNCTION(, char, mock_InterlockedOr8, volatile char*, destination, char, value);
+MOCKABLE_FUNCTION(, LONG, mock_InterlockedXor, volatile LONG*, destination, LONG, value);
+MOCKABLE_FUNCTION(, SHORT, mock_InterlockedXor16, volatile SHORT*, destination, SHORT, value);
+MOCKABLE_FUNCTION(, LONG64, mock_InterlockedXor64, volatile LONG64*, destination, LONG64, value);
+MOCKABLE_FUNCTION(, char, mock_InterlockedXor8, volatile char*, destination, char, value);
+#undef ENABLE_MOCKS
+
+#ifdef __cplusplus
+}
+#endif
+
+BEGIN_TEST_SUITE(interlocked_win32_unittests)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+    ASSERT_ARE_EQUAL(int, 0, umock_c_init(on_umock_c_error));
+    ASSERT_ARE_EQUAL(int, 0, umocktypes_windows_register_types());
+    REGISTER_UMOCK_ALIAS_TYPE(SHORT, int16_t);
```

Comment:
>REGISTER_UMOCK_ALIAS_TYPE(SHORT, int16_t); [](start = 4, length = 42)

hmmm... looks like umocktypes_windows_register_types should do that for you. Would you mind filing a bug against umock_c about this type?